src/gui/basegraphgl.cpp

```diff
@@ -97,6 +98,7 @@ void BaseGraphGL::setup(AbstractGraph* abstractGraph, AttributesScope nodeAttrsS
     m_nodeAttrsScope = nodeAttrsScope;
     setupInspector();
     updateCache();
+    m_attrs = m_abstractGraph->node(0).attrs();
```

Comment:
Is called only after the graph model is initialized so it shouldn't be a problem since we already check whether the number of nodes is not zero. But this must change if we allow to add nodes before generating a graph model

Reply:
nope, this function can also be called with a null abstractGraph; in this scenario, this call would cause a crash. For instance, see `BaseGraphGL::slotRestarted()`:
https://github.com/evoplex/evoplex/blob/28fe85f110f17f2fa02779002e6383a9d058aea5/src/gui/basegraphgl.cpp#L298

could you fix this line please? (i.e., only assign the `attrs()` if `m_abstractGraph` is not null and if `!m_abstractGraph->nodes().empty()`)